draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -734,42 +779,48 @@ It is up to the Holder how to maintain the mapping between the Disclosures and t
 
 ## Processing by the Holder  {#holder_verification}
 
-If a Key Binding JWT is received by a Holder, the SD-JWT SHOULD be rejected.
+The Issuer MUST provide the Holder and SD-JWT, not an SD-JWT-KB.  If the Holder
+receives an SD-JWT-KB from the Issuer, it SHOULD be rejected.
 
 For presentation to a Verifier, the Holder MUST perform the following (or equivalent) steps:
 
  1. Decide which Disclosures to release to the Verifier, obtaining proper End-User consent if necessary.
- 2. Assemble the SD-JWT for presentation, including the Issuer-signed JWT and the selected Disclosures (see (#data_formats) for the format).
- 3. If Key Binding is required, create a Key Binding JWT ((#kb-jwt)) and add it to the SD-JWT for presentation.
- 4. Send the SD-JWT for presentation to the Verifier.
+ 2. Assemble the SD-JWT, including the Issuer-signed JWT and the selected Disclosures (see (#data_formats) for the format).
+ 3. If Key Binding is not required, send the SD-JWT to the Verifier.
+ 4. Create a Key Binding JWT covering the SD-JWT.
+ 5. Assemble the SD-JWT-KB, including the SD-JWT and the Key Binding JWT.
+ 6. Send the SD-JWT-KB to the Verifier.
 
 ## Verification by the Verifier  {#verifier_verification}
 
-Upon receiving a Presentation, in addition to the checks outlined in (#sd_jwt_verification), Verifiers need to ensure that
+Upon receiving a presentation from a Holder, in the form of either an SD-JWT or
+an SD-JWT-KB, in addition to the checks outlined in (#sd_jwt_verification), Verifiers MUST ensure that
 
- * if Key Binding is required, the Key Binding JWT is signed by the Holder and valid.
+ * if Key Binding is required, then the Holder has provided an SD-JWT-KB, and the Key
+   Binding is signed by the Holder and valid.
 
 To this end, Verifiers MUST follow the following steps (or equivalent):
 
 1. Determine if Key Binding is to be checked according to the Verifier's policy
    for the use case at hand. This decision MUST NOT be based on whether
    a Key Binding JWT is provided by the Holder or not. Refer to (#key_binding_security) for
    details.
-2. Process the SD-JWT as defined in (#sd_jwt_verification).
-3. If Key Binding is required:
+2. If Key Binding is required and the Holder has provided an SD-JWT, the Verifier MUST reject the presentation.
+2. If the Holder has provided an SD-JWT-KB, parse it into an SD-JWT and a Key Binding JWT.
```

Comment:
steps 4.1 should come before steps in lines 808 and 809 (there are two steps 2 btw). in other words, the verifier need to know what mechanism it is using for key binding before rejecting because there is no KB JWT